govcd/metadata.go

```diff
@@ -141,7 +141,7 @@ func (vm *VM) DeleteMetadataEntry(key string) error {
 // and returns the task.
 // Deprecated: Use VM.DeleteMetadataEntryWithDomainAsync instead
 func (vm *VM) DeleteMetadataEntryAsync(key string) (Task, error) {
-	return deleteMetadata(vm.client, vm.VM.HREF, key, false)
+	return deleteMetadata(vm.client, vm.VM.HREF, "", key, false)
```

Comment:
Should this not have a `vm.VM.Name`?

Reply:
Mmm yes, I wonder why tests didn't pick up this one. I need to revisit them.

Reply:
Ah, this is a deprecated method, so the filter is not used here, so that explains it. But anyway, for completeness it's nice to have it fixed.